clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -387,6 +388,13 @@ synchronized void maybeSeekUnvalidated(TopicPartition tp, long offset, OffsetRes
         return new HashSet<>(this.assignment.partitionSet());
     }
 
+    /**
+     * @return a modifiable copy of the currently assigned partitions as a list
+     */
+    public synchronized List<TopicPartition> assignedPartitionsList() {
```

Comment:
Did you add this just to avoid an additional copy?

By the way, it's a little weird to indicate that the result is modifiable (I know you were just copying). That seems to suggest that the changes are propagated back to `SubscriptionState`.

Reply:
Yeah, it seemed awkward to be copying twice. And idk, to me "a modifiable copy" means you can safely modify the copy. I guess that's sort of inherent in the definition of copy, but I don't interpret it as meaning modifications to the copy will affect the original

Reply:
Ok, I guess the word "copy" is clear enough.